client/src/main/java/org/spine3/base/error/IllegalConversionArgumentException.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.base.error;
+
+/**
+ * Thrown to indicate that a method has been passed an illegal or
+ * inappropriate argument during the conversion from one type to another.
+ *
+ * @author Illia Shepilov
+ * @see ConversionException
+ */
+public class IllegalConversionArgumentException extends RuntimeException {
+
+    private static final long serialVersionUID = 1L;
+
+    private final ConversionException conversionException;
+
+    public IllegalConversionArgumentException(ConversionException conversionException) {
```

Comment:
We can use `cause` parameter, but it is `Throwable`.
So, `ConversionException` is passed through constructor and is accessible through `getConversionException()` method to avoid `instanceof` and cast.